### PR TITLE
fix(s3): disable client retries when congestion control is enabled

### PR DESCRIPTION
**Pull request by ashwanthgoli**
**What this PR does / why we need it**:

With congestion control enabled, we want the cc controller to handle all retries. It is necessary to disable retries performed by the object client.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: pkg/storage/factory.go [hunk 1]
```diff
@@ -295,7 +295,7 @@ type Config struct {
 	DisableBroadIndexQueries bool         `yaml:"disable_broad_index_queries"`
 	MaxParallelGetChunk      int          `yaml:"max_parallel_get_chunk"`
 
-	UseThanosObjstore bool          `yaml:"use_thanos_objstore" doc:"hidden`
+	UseThanosObjstore bool          `yaml:"use_thanos_objstore" doc:"hidden"`
```
Comment (trevorwhitney) [inline pkg/storage/factory.go:298]: I'm confused how this is related?

**Review by trevorwhitney — APPROVED** [review]
LGTM, but I'm confused how the `use_thanos_objstore` change is related?

File: pkg/storage/factory.go (hunk 1 above)
Comment (ashwanthgoli) [reply pkg/storage/factory.go:298]: it's not, my editor corrected the incorrect tag

Comment (ashwanthgoli) [reply pkg/storage/factory.go:298]: i think i'll keep this change here instead of raising another pr for this small correction :)

**Pull request merged**